### DISCUSSION TITLE
Suggestion: Stop using NHSpecificTest

### DISCUSSION THREAD

**Discussion by gliljas** [nhibernate/nhibernate-core#2562] [General]
IMO, the NHSpecificTest structure comes from a time before Git, pull requests etc. and before NHibernate diverged from Hibernate. It encourages discrepancy and unmaintainability. Of course, good guidelines on how and where to write tests will be necessary, and ideally all the tests in NHSpecificTest should be incorporated elsewhere.

**Comment by hazzik** [comment]
I could not agree more.

**Comment by bahusoid** [comment]
Well expecting issue reporters to investigate huge test codebase in search of proper folder sounds a bit strange to me. Not user friendly suggestion. They simply won't do it. And when issue is created the area could be unknown. 

So this suggestion looks unrealistic to me. Yeah we can analyze existing tests and move them in proper area (who wants to do it?)  but we shouldn't mandate it for issue reporters.

And it's duplicate of https://github.com/nhibernate/nhibernate-core/issues/1409 :)

**Reply by gliljas** [reply]
Ouch... well, a brand new discussion thingy isn't a bad thing.

If it's hard to find the proper place, I'd say that's a failure of the existing test code base. I rarely find it hard to add unit tests to other open source projects. And the PR reviews are there for a reason. A suggestion to rename/move a test should be seen as a friendly thing.

**Reply by bahusoid** [reply]
I agree it's a good thing to place tests right in proper folder from start. But IMO if issue is unclear or no one is willing to investigate it properly right away (most common scenario here) - it's better to commit it in NHSpecificFolder than keep it hang in open PRs or commit it to some unrelated folder.

P.S. Are you actually going to review PRs and suggest proper place for tests and/or restructure existing tests? :) Or it's just another brand new discussion?

**Reply by gliljas** [reply]
The guidelines are needed first.
